src/Symfony/Component/Translation/Bridge/Crowdin/CrowdinProvider.php

```diff
@@ -0,0 +1,425 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Bridge\Crowdin;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Translation\Dumper\XliffFileDumper;
+use Symfony\Component\Translation\Exception\ProviderException;
+use Symfony\Component\Translation\Loader\LoaderInterface;
+use Symfony\Component\Translation\Provider\ProviderInterface;
+use Symfony\Component\Translation\TranslatorBag;
+use Symfony\Component\Translation\TranslatorBagInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @author Andrii Bodnar <andrii.bodnar@crowdin.com>
+ *
+ * In Crowdin:
+ *  * Filenames refer to Symfony's translation domains;
+ *  * Identifiers refer to Symfony's translation keys;
+ *  * Translations refer to Symfony's translated messages
+ *
+ * @experimental in 5.3
+ */
+final class CrowdinProvider implements ProviderInterface
+{
+    /** @var HttpClientInterface */
+    private $client;
+
+    /** @var LoaderInterface */
+    private $loader;
+
+    /** @var LoggerInterface */
+    private $logger;
+
+    /** @var XliffFileDumper */
+    private $xliffFileDumper;
+
+    /** @var string */
+    private $defaultLocale;
+
+    /** @var string */
+    private $endpoint;
+
+    /** @var string */
+    private $projectId;
+
+    /** @var HttpClientInterface */
+    private $filesDownloader;
+
+    public function __construct(HttpClientInterface $client, LoaderInterface $loader, LoggerInterface $logger, XliffFileDumper $xliffFileDumper, string $defaultLocale, string $endpoint, int $projectId, HttpClientInterface $filesDownloader)
+    {
+        $this->client = $client;
+        $this->loader = $loader;
+        $this->logger = $logger;
+        $this->xliffFileDumper = $xliffFileDumper;
+        $this->defaultLocale = $defaultLocale;
+        $this->endpoint = $endpoint;
+        $this->projectId = $projectId;
+        $this->filesDownloader = $filesDownloader;
```

Comment:
Why to you need a different HTTP Client here?

Reply:
I tried to use the HTTP client passed to this Provider, but it already configured with Crowdin base URI and Authorization header and it caused the Bad Request error because the URL from the response is not a Crowdin link, it's an S3 link

then @stof [suggested](https://github.com/symfony/symfony/pull/40947#discussion_r622497857) using a different HTTP Client

Reply:
Ok, I didn't notices that links was from S3, not from Crowdin